tests/ImageConfigTest.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Spatie\Image\Test;
+
+use Spatie\Image\Image;
+use Spatie\Image\Manipulations;
+use Spatie\Image\Exceptions\DefectiveConfiguration;
+
+class ImageConfigTest extends TestCase
+{
+    /** @test */
+    public function it_can_modify_an_image_while_setting_temporary_path_by_static_method()
+    {
+        $targetFile = $this->tempDir->path('conversion.jpg');
+        Image::setTemporaryDirectory(__DIR__.'/temp_conf');
+        Image::load($this->getTestJpg())
+            ->manipulate(function (Manipulations $manipulations) {
+                $manipulations
+                    ->blur(50);
+            })
+            ->save($targetFile);
+
+        $this->assertFileExists($targetFile);
+    }
+
+    /** @test */
+    public function it_will_throw_an_error_if_tempdir_corrupt()
+    {
+        $this->expectException(DefectiveConfiguration::class);
+        $targetFile = $this->tempDir->path('conversion.jpg');
+        Image::setTemporaryDirectory('/user/willmostprobablynotexistandcreatable');
+        Image::load($this->getTestJpg())
+            ->manipulate(function (Manipulations $manipulations) {
+                $manipulations
+                    ->blur(50);
+            })
+            ->save($targetFile);
+    }
+
+    protected function tearDown()
+    {
+        Image::setTemporaryDirectory(__DIR__.'/');
```

Comment:
If setting the temp directory works in a non-static way, this won't be necessary anymore.

Reply:
@freekmurze Thanks for the review. Please allow me one question though : if the method for setting the temporary directory is not static, there is no way for me setting it if i use libraries like laravel-medialibrary. So i will have to make an other pull request there too, because every Image::load also must be followed by the directory-setting method right ?

If this is generally the right way to do it to keep things decoupled / clean , then i am okay with that, i just want to know if you were aware of my origin purpose (that is: setting the tempdir while using laravel-medialibrary) which will not be solved by this pull-request then.

Thanks again. Looking forward to your answer.

Reply:
When this PR gets merged and tagged, send a PR to medialib to make use of these changes.